### PR TITLE
[SPARK-34935][SQL] CREATE TABLE LIKE should respect the reserved table properties

### PR DESCRIPTION
**Pull request by yaooqinn**
### What changes were proposed in this pull request?

CREATE TABLE LIKE should respect the reserved properties of tables and fail if specified, using `spark.sql.legacy.notReserveProperties` to restore.

### Why are the changes needed?

Make DDLs consistently treat reserved properties

### Does this PR introduce _any_ user-facing change?

YES, this is a breaking change as using `create table like` w/ reserved properties will fail.

### How was this patch tested?

new test

### CODE REVIEW

**Comment by yaooqinn** [comment]
cc @cloud-fan @maropu thanks for reviewing, I guess we can make a document for reserved properties for DDL pages to refer

File: docs/sql-migration-guide.md [hunk 1]
```diff
@@ -73,6 +73,8 @@ license: |
 
   - In Spark 3.2, the timestamps subtraction expression such as `timestamp '2021-03-31 23:48:00' - timestamp '2021-01-01 00:00:00'` returns values of `DayTimeIntervalType`. In Spark 3.1 and earlier, the type of the same expression is `CalendarIntervalType`. To restore the behavior before Spark 3.2, you can set `spark.sql.legacy.interval.enabled` to `true`.
 
+  - In Spark 3.2, `CREATE TABLE .. LIKE ..` command can not use reserved properties. You need their specific clauses to specify them, for example, `CREATE TABLE test1 LIKE test LOCATION 'some path'`. You can set `spark.sql.legacy.notReserveProperties` to `true` to ignore the `ParseException`, in this case, these properties will be silently removed, for example: `TBLPROPERTIES('owner'='yao')` will have no effect. In Spark version 3.1 and below, the reserved properties are only affect other DDLs except for this one, for example, `TBLPROPERTIES('location'='/tmp')` does not change the location of the table but only create a headless property just like `'a'='b'`.
+
```
Comment (cloud-fan) [inline docs/sql-migration-guide.md:77]: `the reserved properties are only affect other DDLs except for this one` what does it mean? What's "this one"?

Comment (yaooqinn) [reply docs/sql-migration-guide.md:77]: It's a typo, let me fix it:)

**Review by cloud-fan — APPROVED** [review]

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM.

**Review by maropu — APPROVED** [review]

**Comment by maropu** [comment]
Thanks! Merged to master.

**Pull request closed without merging**